container_api/src/config.rs

```diff
@@ -0,0 +1,223 @@
+use holochain_core_types::{
+    entry::agent::{Agent, Identity},
+    error::{HcResult, HolochainError},
+    json::JsonString,
+};
+use holochain_dna::Dna;
+use serde::Deserialize;
+use std::{convert::TryFrom, fs::File, io::prelude::*};
+
+#[derive(Deserialize)]
+pub struct Configuration {
+    #[serde(rename = "agent")]
+    agents: Option<Vec<AgentConfiguration>>,
+    #[serde(rename = "dna")]
+    dnas: Option<Vec<DNAConfiguration>>,
+    #[serde(rename = "instance")]
+    instances: Option<Vec<InstanceConfiguration>>,
+    #[serde(rename = "interface")]
+    interfaces: Option<Vec<InterfaceConfiguration>>,
+    #[serde(rename = "bridge")]
+    bridges: Option<Vec<Bridge>>,
+}
+
+#[derive(Deserialize, Clone)]
+pub struct AgentConfiguration {
+    id: String,
+    key_file: Option<String>,
+}
+
+impl Into<Agent> for AgentConfiguration {
+    fn into(self) -> Agent {
+        Agent::from(Identity::from(self.id))
+    }
+}
+
+#[derive(Deserialize)]
+pub struct DNAConfiguration {
+    id: String,
+    file: String,
+    hash: String,
+}
+
+impl Into<HcResult<Dna>> for DNAConfiguration {
```

Comment:
Feels a bit heavy to include file IO in an `into()` conversion but I'm not versed enough to know how people do it.

Reply:
Inherited that impl from @AshantiMutinta.
I do think this is adequate for an Into impl. What do you say, Ashanti?

On a related note, this could be a TryInto, no?

Reply:
Definitely could be a tryinto, think it is okay to have the IO in there as the struct is a configuration struct but open to other ideas definitely @lucksus @maackle